elasticache/redis/redis.go

```diff
@@ -0,0 +1,139 @@
+	vpcSecurityGroupIDs = append(vpcSecurityGroupIDs, sg.SecurityGroupId())
+
+	// build up the secrets manager
+	sess, err := session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create AWS session: %w", err)
+	}
+
+	sm := secretsmanager.New(sess)
+
+	// generate the token
+	token, err := util.GenerateToken(24)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate token: %w", err)
+	}
+
+	// store the token
+	secretOutput, err := sm.CreateSecret(&secretsmanager.CreateSecretInput{
+		Name:         jsii.String(props.ClusterName + "-token"),
+		SecretString: jsii.String(token),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create token secret: %w", err)
+	}
+
+	// create the Redis cluster
+	// it might seem like creating a replication group is not the same as creating a cluster
+	// but actually it creates the cluster and the replication group in one go
+	redisRG := awselasticache.NewCfnReplicationGroup(stack, jsii.String(props.ClusterName), &awselasticache.CfnReplicationGroupProps{
+		ReplicationGroupId:          jsii.String(props.ClusterName),
+		ReplicationGroupDescription: jsii.String("Acorn created Redis replication group"),
+		Engine:                      jsii.String("redis"),
+		CacheNodeType:               jsii.String("cache.t4g.micro"),
+		// this says num clusters but with cluster mode disabled its actually num nodes
+		// also, the terminology is confusing. We're creating an elasticache cluster but not a Redis cluster.
+		NumCacheClusters:         jsii.Number(3),
+		AutomaticFailoverEnabled: jsii.Bool(true),
+		TransitEncryptionEnabled: jsii.Bool(true),
+		CacheSubnetGroupName:     subnetGroup.CacheSubnetGroupName(),
+		SecurityGroupIds:         &vpcSecurityGroupIDs,
+		AuthToken:                jsii.String(token),
+		Port:                     jsii.Number(6379),
+	})
+
+	// output the cluster details
+	awscdk.NewCfnOutput(stack, jsii.String("clustername"), &awscdk.CfnOutputProps{
+		Value: jsii.String(props.ClusterName),
+	})
+	awscdk.NewCfnOutput(stack, jsii.String("address"), &awscdk.CfnOutputProps{
+		Value: redisRG.AttrPrimaryEndPointAddress(),
+	})
+	awscdk.NewCfnOutput(stack, jsii.String("port"), &awscdk.CfnOutputProps{
+		Value: redisRG.AttrPrimaryEndPointPort(),
+	})
+	awscdk.NewCfnOutput(stack, jsii.String("tokenarn"), &awscdk.CfnOutputProps{
+		Value: secretOutput.ARN,
+	})
+
+	return stack, nil
+}
+
+func main() {
+	defer jsii.Close()
+
+	app := common.NewAcornTaggedApp(nil)
+
+	stackProps := &redisStackProps{
+		StackProps: *common.NewAWSCDKStackProps(),
+	}
+	stackProps.VpcID = common.GetVpcID()
+
+	err := common.NewConfig(stackProps)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to create config")
+	}
+
+	id, err := util.GenerateID(8)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to generate cluster ID")
+	}
+
+	stackProps.ClusterName = stackProps.ClusterName + "-" + id
```

Comment:
Why is this adding an ID? Can it be the ACORN_EXTERNAL_ID || user specified name?

Reply:
Now using `ACORN_PROJECT` + `ACORN_NAME`as we discussed.